microservices-orders/run-ksql.sh

```diff
@@ -1,7 +1,14 @@
 #!/bin/bash
 
-echo "Sleeping 300 seconds until ksqlDB server starts and topics have data"
-sleep 300
+echo -e "\nWaiting for ksqlDB and topics to be available prior to running ksqlDB statements\n"
+while [[ $(curl -s http://ksqldb-server:8088/info) != *RUNNING* \
+             ||  $(curl -s http://schema-registry:8081/subjects) != *customers* \
+             ||  $(curl -s http://schema-registry:8081/subjects) != *orders* ]]
+do 
+    sleep 5
+    echo "Sleeping for 5 seconds, waiting for ksqlDB and topics to be available"
+done
+
```

Comment:
Nice change @awalther28 .  What do you think about leveraging the existing, and available `retry` function?  It has 'abort' capability.  Maybe you could wrap up the checks into a function inside this script and then invoke it with `retry` or figure out a slick way to pass in these calls to the `retry` directly.  There is an example of `retry` usage in the `read-topics-docker.sh` script and it's defined in `utils/helper.sh`

Reply:
Or maybe that's not available inside the docker container.... so that might not work.  If not, then I'm wondering if we should add a max wait equivilant here?

Reply:
Another consideration for you, if you decided to break this into a function, you could remove the redundant curl calls to SR and stuff the result into a variable, then check the value of the variable instead of invoke `curl` twice for the same info.  There also might be a solution using a REGEX but that could be more work than it's worth.

Reply:
@rspurgeon I have applied some of the changes you suggested: namely creating functions, limiting curl usage, and leveraging `utils/helper.sh retry`. Bash is not a strong suit of mine, definitely open to any suggestions you have to make things better.

Reply:
@rspurgeon bump 🙂

Reply:
nice :-)